### DISCUSSION TITLE
Please Add A Square Ratio Selection ability

### DISCUSSION THREAD

**Discussion by CodexEnt** [flameshot-org/flameshot#3072] [Q&A]
Please add a shortcut key to the rectangle area selection to lock it at a square ratio.

This is a popular ratio due to Instagram and other programs that require square images.

Such a simple thing, but if Flameshot adds this, would set it apart from other software, still haven't run into one that doesn't just have an extra ctrl or alt or shift context key while dragging that will limit it to equal width and height while resizing.

If I missed something and we can do this already, would appreciate it, can't find the feature :D

**Comment by mmahmoudian** [comment]
@CodexEnt @AdamUhh have you tried holding down <kbd>Ctrl</kbd> while changing size of the region?

We try to follow Inkscape's keybindings as much as we can. For this reason
- holding <kbd>Ctrl</kbd> would preserve the aspect ratio
- holding <kbd>Shift</kbd> would do symmetric resizing.

**Reply by AdamUhh** [reply]
Hey,
That isn't working for me for some reason, and after some testing, I found that the Ctrl key is acting all weird
I can save and copy the selection (Ctrl+S/Ctrl+C) but trying to "Symmetrically resize selection 2px" (Ctrl + Shift + Arrow) doesn't work at all, which is probably why pressing the key doesn't do anything for me.
I'll probably put up a post on Issues if I can't figure this out :/

**Reply by mmahmoudian** [reply]
@AdamUhh what is your OS? Is it macOS by any chance?

**Reply by AdamUhh** [reply]
Nopee, its Ubuntu 22.04

**Reply by mmahmoudian** [reply]
I just realized you said "Ctrl+Shift+arrow". What I said in my previous post is to hold Ctrl while changing the size. At the moment changing size is only via mouse click-and-drag or the handles

**Comment by CodexEnt** [comment]
Workflow:
1) Click on icon on taskbar to activate Flameshot, draw out initial shape of capture.
Result is always going to be a non-square somehow.
* I would like to be able to select a square region (equal height and width), and ability to scale (with locked SQUARE aspect ratio) to desired size.
Usage: For Instagram style square images and for other purposes that require square images (Many shop websites from Shopify to Etsy)

Current Issues:
1) Cannot obtain a SQUARE ratio without great effort (using the Width*Height prompt circle in interface)
2) Even if I could, when I want to scale UP or DOWN, maintaining a SQUARE ratio,  the following methods do NOT lock the ratio

CTRL + mousedrag on corner does NOT lock the ratio while dragging, but it does scale evenly, therefore disrupting the SQUARE ratio.
CTRL + SHIFT + Mousedrag on corner SCALES the selection area EVENLY, larger and smaller, but does not lock the ratio (height and width) you started with.

mousedrag, CTRL + mousedrag on side and top handles only moves those individual handles
CTRL + SHIFT + Mousedrag on side and top handles scales on X and Y

Suggestion?  Flameshot activate -> hold CTRL down + drag = select with even aspect (SQUARE) immediately

Similar behaviour to "Rectangle Marquee Tool (M)" in Photoshop:
MouseDown+ Drag: Drag selection for loose rectangle solution  (this is  default Flameshot selection)
MouseDown + Drag + Ctrl + Shift: Make selection in SQUARE shape 
MouseDown + Drag + Ctrl + Shift + Alt: Make selection in SQUARE shape from center of mouse position

**Reply by mmahmoudian** [reply]
Already implemented. We do closely follow Inkscape keyboard shortcut scheme. So holding Ctrl and resizing the selected region would keep the aspect ratio.


As the matter of fact, I just tried it:

```sh
flameshot gui --region 100x100+100+100
```

And then hold Ctrl and drag. If you have the dimensions showing, you will see that the height and width stay equal.

As for drawing a square annotation, hold Ctrl **before** clicking and dragging. Circle follow the same pattern.

**Reply by CodexEnt** [reply]
Windows 10 operating system, as my last post suggested, the expected behaviour does not happen.

1) Click Flameshot Icon (Preloaded) on taskbar
2) Hold CTRL  + DRAG and select does NOT offer any locked size ratio
- Width and Height do not stay equal to or represent a square ratio
3) Hold SHIFT + DRAG  and the selection scales from the center, but does not offer any locked size ratio
- Width and Height do not stay equal to or represent a square ratio

Thanks

**Reply by mmahmoudian** [reply]
This feature (fixing aspect ratio upon holding Ctrl) was implemented in #2931 which was merged 2022-10-03. If you have installed Flameshot from a build generated **after** this date, then you have it, otherwise you don't. We have not had a release for about a year, so if you want this feature along with other improvements, your only choice is to download the nightly build and install from that:

https://flameshot.org/docs/installation/development-build/

In the nighly builds, by holding <kbd>Ctrl</kbd> you can keep the aspect ratio static while changing the side of the selection and also while drawing objects.

Therefore, this feature request is duplicate of #1132

**Reply by andrewanswer** [reply]
> This feature (fixing aspect ratio upon holding Ctrl) was implemented in #2931 which was merged 2022-10-03. If you have installed Flameshot from a build generated **after** this date, then you have it, otherwise you don't. We have not had a release for about a year, so if you want this feature along with other improvements, your only choice is to download the nightly build and install from that:
> 
> https://flameshot.org/docs/installation/development-build/
> 
> In the nighly builds, by holding Ctrl you can keep the aspect ratio static while changing the side of the selection and also while drawing objects.
> 
> Therefore, this feature request is duplicate of #1132

Hi,

holding CTRL does not lock aspect ratio. I tried it accurately on my Ubuntu 24.04 LTS, Flameshot 14.0.0

For example, select 600x200 area and extend it using CTRL to 900x300. You can't do it! The ratio slowly degrade to 1:1 during resize.

**Reply by mmahmoudian** [reply]
For the record, this is now fixed in #4651

**Comment by anasouardini** [comment]
## Env
- OS: Debian 12.8
- Flameshot version: Flameshot v12.1.0 (Debian 12.1.0-2) and Compiled with Qt 5.15.8

## behavior
- When holding `Shift` symetric resizing works like a sharm
- When  holding `Ctrl` it doesn't preserve the ratio

## Question
- If it's a dependency problem, would an AppImage make it work? if so, where can I get an AppImage for it?

**Reply by mmahmoudian** [reply]
The AppImage or the Flatpak should not change how Ctrl is passed to Flameshot. What is your window manager?

**Comment by CodexEnt** [comment]
Windows 10

Shift does not LOCK the aspect ratio of the window selection.

It DOES mirror the WIDTH of the selection when pulling left or right (Symmetry Width)
It DOES mirror the HEIGHT of the selection when pulling up or down. (Symmetry Height)

The symmetry pulls from the center of the selection

---

Ctrl does nothing.

Flameshot v12.1.0 (96c2c82e)
Compiled with Qt 5.15.2
winnt: 10.0.19045
windows: 10

**Comment by markbrewster** [comment]
Linux Mint

CTRL does not appear to lock the aspect ratio when selecting as per earlier reports.

Flameshot v12.1.0 (Debian 12.1.0-2build2)
Compiled with Qt 5.15.12
linux: 6.17.0-14-generic
linuxmint: 22.3

**Reply by borgmanJeremy** [reply]
V12 is outdated and not supported. You can get v13 from our release page our wait a few weeks for v14 which will fix lots of bugs.

**Reply by markbrewster** [reply]
My bad - can confirm v13 square capture with CTRL works on Linux Mint. I was using a vesion from an old repo. Thanks for the quick reply @borgmanJeremy
